core/chains/evm/txmgr/strategies.go

```diff
@@ -45,22 +49,23 @@ var _ TxStrategy = DropOldestStrategy{}
 // DropOldestStrategy will send the newest N transactions, older ones will be
 // removed from the queue
 type DropOldestStrategy struct {
-	subject   uuid.UUID
-	queueSize uint32
+	subject      uuid.UUID
+	queueSize    uint32
+	queryTimeout time.Duration
 }
 
 // NewDropOldestStrategy creates a new TxStrategy that drops the oldest transactions after the
 // queue size is exceeded.
-func NewDropOldestStrategy(subject uuid.UUID, queueSize uint32) DropOldestStrategy {
-	return DropOldestStrategy{subject, queueSize}
+func NewDropOldestStrategy(subject uuid.UUID, queueSize uint32, queryTimeout time.Duration) DropOldestStrategy {
+	return DropOldestStrategy{subject, queueSize, queryTimeout}
 }
 
 func (s DropOldestStrategy) Subject() uuid.NullUUID {
 	return uuid.NullUUID{UUID: s.subject, Valid: true}
 }
 
 func (s DropOldestStrategy) PruneQueue(q pg.Queryer) (n int64, err error) {
-	ctx, cancel := pg.DefaultQueryCtx()
+	ctx, cancel := context.WithTimeout(context.Background(), s.queryTimeout)
```

Comment:
Same here... maybe too far out of the way for this one, but ideally, txmgr should have passed a child of its context on to `NewQueueingTxStrategy`, which hangs onto it and uses it here.

Reply:
This is also not a change in behavior.

It may be the case that an upstream context should be included here, but if so it should be an explicit method parameter from the caller of `PruneQueue`, not passed along from the caller of the constructor. However, IIUC the `Queryer` already handles that. Either way, out of scope for this PR.